### PR TITLE
Add documentation on how to profile OpenShift

### PR DESCRIPTION
**Pull request by smarterclayton**
[merge]

### CODE REVIEW

File: HACKING.md [hunk 1]
```diff
@@ -319,3 +319,56 @@ OpenShift and Kubernetes integrate with the [Swagger 2.0 API framework](http://s
...
+
+OpenShift integrates the go `pprof` tooling to make it easy to capture CPU and heap dumps for running systems.  The following modes are available for the `openshift` binary (including all the CLI variants):
+
+* `OPENSHIFT_PROFILE` environment variable:
+  * `cpu` - will start a CPU profile on startup and write `./cpu.pprof`.  Contains samples for the entire run at the native sampling resolution (100hz). Note: CPU profiling for Go does not currently work on Mac OS X - the stats are not correctly sampled
+  * `mem` - generate a running heap dump that tracks allocations to `./mem.pprof`
+  * `web` - start the pprof webserver in process at http://127.0.0.1:6060/debug/pprof (you can open this in a browser)
+
+    # start the server in CPU profiling mode
+    $ OPENSHIFT_PROFIE=cpu sudo ./_output/local/go/bin/openshift start
```
Comment (stevekuznetsov) [inline HACKING.md]: Not in `bash` formatting, `PROFILE` misspelled. @smarterclayton

**Comment by danmcp** [comment]
[test]

**Comment by stevekuznetsov** [comment]
travis deeming this a failure is hilarious

**Comment by mfojtik** [comment]
thanks for this @smarterclayton :)

**Pull request merged**